### ISSUE TITLE
insertMany type is always never[]

### ISSUE THREAD

**Issue by jessekrubin** [oramasearch/orama#542]
### Describe the bug

When I try to use insertMultiple I ALWAYS get xyz is not assignable to never[]

### To Reproduce

Try using insertMultiple and then running tsc. Can make a repro later. Currently at park with dogs. 

### Expected behavior

Types should infer correctly. 

### Environment Info

```bash
os: windows/ubuntu/rhel
Node: 18.18.1
Orama: 2.0.0-beta4
```

### Affected areas

Data Insertion

**Comment by rosszurowski** [comment]
I just ran into this too. Looking at [past issues](https://github.com/oramasearch/orama/issues/538#issuecomment-1786167651), it looks like you need to use `as const` for the typings to work.

```
const db = await create({
  schema: {
    id: "string",
    title: "string",
    notes: "string",
  } as const, // the as const here is important
})
```

It looks like it's covered in the [Typescript documentation page](https://docs.oramasearch.com/open-source/usage/typescript), but I didn't even notice they had a Typescript page, and instead was just relying on the [create docs page](https://docs.oramasearch.com/open-source/usage/create).

**Comment by jessekrubin** [comment]
@rosszurowski that semi-works for me, but I get some weird deviation between the return-type of search and the schema due to the `readonly`s from the `as const`. Fixable with some type-gymnastics tho.

**Comment by micheleriva** [comment]
@rosszurowski good call. We will fix the docs adding TypeScript specifications on other pages 🙏

**Comment by jessekrubin** [comment]
@micheleriva how can I make orama fuzzier? I am currently testing it out as a replacement for [fzf-for-js](https://github.com/ajitid/fzf-for-js) which fuzzy matches more aggressively

**Comment by jacob-8** [comment]
> @rosszurowski good call. We will fix the docs adding TypeScript specifications on other pages 🙏

Did this ever happen? I actually came across the `as const` in https://docs.oramasearch.com/open-source/usage/typescript and still didn't realize it was my problem because I recently ran into a "never" issue using Supabase because I did not have `strictNullChecks` set to `true`. So that took me down the wrong route to fix things until coming to this issue. Consider mentioning it in the Getting Started, Create, and Insert pages.
